extensions/grunt/src/main.ts

```diff
@@ -259,7 +278,7 @@ class TaskDetector {
 		if (folders) {
 			for (let folder of folders) {
 				if (!this.detectors.has(folder.uri.toString())) {
-					let detector = new FolderDetector(folder);
+					let detector = new FolderDetector(folder, findGruntCommand(folder.uri.fsPath));
 					if (detector.isEnabled()) {
 						this.detectors.set(folder.uri.toString(), detector);
```

Comment:
This line should be outside of the `isEnabled` check.

Reply:
And above, on line 272 the `isEnabled` check should be removed. We want to make sure the detector is always disposed and deleted now that it can still be used when the setting isn't enabled.